test/e2e/toolchaincluster_test.go

```diff
@@ -46,6 +52,18 @@ func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wa
 		require.Equal(t, current.Name, secret.Labels[toolchainv1alpha1.ToolchainClusterLabel], "the secret of the ToolchainCluster %s is not labeled", client.ObjectKeyFromObject(&current))
 	})
 
+	// NOTE: this checks that there is the migration step in place that converts the connection details for the cluster that is currently
+	// spread over the secret and ToolchainCluster is also set as a kubeconfig inside the secret. Once we migrate to create the ToolchainClusters
+	// based on the secrets, we will no longer require this migration step and this test will be removed.
+	t.Run("kubeconfig is generated from the connection details", func(t *testing.T) {
+		secret := &corev1.Secret{}
+		require.NoError(t, await.Client.Get(context.TODO(), client.ObjectKey{Name: current.Spec.SecretRef.Name, Namespace: current.Namespace}, secret))
+
+		// this is enough for the test here. The correctness of the contained kubeconfig is checked in the unit tests of the controller in toolchain-common.
+		_, present := secret.Data["kubeconfig"]
+		assert.Equal(t, kubeConfigMustExist, present)
```

Comment:
There are actually already functions that helps you with verifying the validity of the kubeconfig
https://github.com/codeready-toolchain/toolchain-e2e/blob/6c68589aa96acba09217ee95e9fdc05d0ba19fb0/testsupport/space/spacerequest.go#L115-L149

There is no need to do anything fancy, I would just export the functions and use it here - something like this:
```suggestion
		if kubeConfigMustExist {
			kubeconfig, _ := space.NewKubeClientFromSecret(t, await.Client, current.Spec.SecretRef.Name, current.Namespace)
			space.ValidateKubeClient(t, kubeconfig, current.Labels["namespace"])
		} else {
			secret := &corev1.Secret{}
			require.NoError(t, await.Client.Get(context.TODO(), client.ObjectKey{Name: current.Spec.SecretRef.Name, Namespace: current.Namespace}, secret))

			_, present := secret.Data["kubeconfig"]
			assert.False(t, present)
		}
```
But if it would require more changes  than I assume in the code above, then please disregard my comment and keep it as it is.